builtin/providers/ultradns/resource_ultradns_dirpool.go

```diff
@@ -610,3 +619,78 @@ func mapEncode(rawVal interface{}) map[string]interface{} {
 	s.TagName = "terraform"
 	return s.Map()
 }
+
+// hashRdataDirpool generates a hashcode for an Rdata block from dirpools
+func hashRdataDirpool(v interface{}) int {
+	var buf bytes.Buffer
+	m := v.(map[string]interface{})
+	buf.WriteString(fmt.Sprintf("%s-", m["host"].(string)))
+	buf.WriteString(fmt.Sprintf("%t-", m["all_non_configured"].(bool)))
+
+	if geoInfo, ok := m["geo_info"]; ok {
+		var g map[string]interface{}
+		switch geoInfo := geoInfo.(type) {
+		case []interface{}:
+			if len(geoInfo) >= 1 {
+				g = geoInfo[0].(map[string]interface{})
+			}
+		case []map[string]interface{}:
+			if len(geoInfo) >= 1 {
+				g = geoInfo[0]
+			}
+		}
+		if g != nil {
+			buf.WriteString(fmt.Sprintf("%s-", g["name"].(string)))
+			buf.WriteString(fmt.Sprintf("%t-", g["is_account_level"].(bool)))
+
+			codes := g["codes"].(*schema.Set).List()
+			if len(codes) >= 1 {
+				sort.Slice(codes, func(i, j int) bool { return codes[i].(string) < codes[j].(string) })
+				for _, c := range codes {
+					buf.WriteString(fmt.Sprintf("%s,", c))
+				}
+			}
+		}
+	}
+
+	if ipInfo, ok := m["ip_info"]; ok { //}.([]interface{})
+		var i map[string]interface{}
+		switch ipInfo := ipInfo.(type) {
+		case []interface{}:
+			if len(ipInfo) >= 1 {
+				i = ipInfo[0].(map[string]interface{})
+			}
+		case []map[string]interface{}:
+			if len(ipInfo) >= 1 {
+				i = ipInfo[0]
+			}
+		}
+		if i != nil {
+			buf.WriteString(fmt.Sprintf("%s-", i["name"].(string)))
+			buf.WriteString(fmt.Sprintf("%t-", i["is_account_level"].(bool)))
+
+			ips := i["ips"].(*schema.Set).List()
+			if len(ips) >= 1 {
+				for _, p := range ips {
+					ip := p.(map[string]interface{})
+					if ip["start"] != nil {
+						buf.WriteString(fmt.Sprintf("%s-", ip["start"].(string)))
+					}
+					if ip["end"] != nil {
+						buf.WriteString(fmt.Sprintf("%s-", ip["end"].(string)))
+					}
+					if ip["cidr"] != nil {
```

Comment:
Since your are only building up a hash string, why do you need all these nil checks?

Reply:
I guess because the element may not be initialized or exist

Reply:
Right, they don't all exist at the same time. Using some of these excludes using others in actual configurations.